src/stores/filter.js

```diff
@@ -17,11 +17,12 @@ import {
 import { warn } from '~/utils/console'
 
 export const useFilterStore = defineStore('filter', () => {
-  /** @type {{ filters: import('../store/types').Filters}} */
+  /** @type {import('../store/types').Filters} */
   const filters = reactive(clonedeep(filterData))
-  const searchType = /** @type {import('../store/types').SearchType} */ (
-    ref(ALL_MEDIA)
-  )
+  const searchType =
+    /** @type {import('@nuxtjs/composition-api').Ref<import('../store/types').SearchType>} */ (
+      ref(ALL_MEDIA)
+    )
```

Comment:
It would be more concise to cast `ALL_MEDIA` than the whole ref (can avoid importing `Ref`). That would match more of what you'd do in actual TS too:

```suggestion
  const searchType = ref(/** @type {import('~/store/types').SearchType} */ (ALL_MEDIA))
```